dbus/xdg.c

```diff
@@ -343,6 +361,16 @@ static int handle_notify(sd_bus_message *msg, void *data,
 	}
 	notif->requested_timeout = requested_timeout;
 
+	if (notif->tag) {
+		// Find and replace the existing notfication with a matching tag
+		struct mako_notification *replace_notif = get_tagged_notification(state, notif->tag);
+		if (replace_notif) {
+			wl_list_insert(&replace_notif->link, &notif->link);
+			destroy_notification(replace_notif);
```

Comment:
This doesn't replace the notification, this closes the previous one and opens a new one. This has consequences on e.g. ordering and IDs.

See how `replaces_id` is handled at the top of the function.

Reply:
tl;dr we need to reset the replaced notification, and then to copy over all the fields from `notif` to the replaced notification.

Reply:
Maybe an easier trick would be to override `notif->id` with the replaced notification's, then `wl_list_insert(&replaced->link, &notif->link)` to insert the "new" notif at the right position, then destroy `replaced`.

Reply:
Updated to override the new `notif->id`. It means we lose an ID, but it saves the hassle of copying the new notification into the original.

Reply:
Yup, that sounds reasonable!